integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -62,6 +63,26 @@ def test_multiplication(data_gen):
                 f.col('a') * f.col('b')),
             conf=allow_negative_scale_of_decimal_conf)
 
+# No overflow gens here because we just focus on verifying the fallback to CPU when
+# enabling ansi mode. But overflows will fail the tests because CPU runs raise
+# exceptions.
+_no_overflow_multiply_gens = [
+    ByteGen(min_val = 1, max_val = 10, special_cases=[]),
+    ShortGen(min_val = 1, max_val = 100, special_cases=[]),
+    IntegerGen(min_val = 1, max_val = 1000, special_cases=[]),
+    LongGen(min_val = 1, max_val = 3000, special_cases=[]),
+    float_gen, double_gen,
```

Comment:
nit: I think we can have a follow on issue to re-enable float, double, and decimal processing in ANSI mode.  Each of them handle overflows already in different ways that should be okay.